diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/BooleanExpressionsRule.kt

```diff
@@ -60,22 +68,39 @@ class BooleanExpressionsRule(configRules: List<RulesConfig>) : DiktatRule(
     }
 
     /**
+     * Converts a complex boolean expression into a string representation, mapping each elementary expression to a letter token.
+     * These tokens are collected into [mapOfExpressionToChar].
+     * For example:
+     * ```
+     * (a > 5 && b != 2) -> A & B
```

Comment:
I remembered this trick from first PR, but don't understand why we need this mapping, it's required for `jbool`?

Reply:
Yes, it operates with boolean expressions with simple variables. Complicated things like `a > 5` it will try to parse